src/hdmf/build/objectmapper.py

```diff
@@ -617,7 +617,9 @@ def __convert_string(self, value, spec):
                     def string_type(x):
                         return x.isoformat()  # method works for both date and datetime
                 if string_type is not None:
-                    if spec.shape is not None or spec.dims is not None:
+                    if spec.shape is not None and len(spec.shape) > 1:
+                        ret = [[string_type(item) for item in sublist] for sublist in value]
```

Comment:
This is fine for 2D arrays but I think it will fail for higher-dimensional arrays. Can you add a test for a 3D array? You could convert value to a numpy array and change the type. Or do the conversion recursively on the list elements.

Reply:
Good point. Converting it to a numpy array and changing the type didn't work as well with the isodatetimes so I added a recursive conversion.